odoo/orm/fields.py

```diff
@@ -700,15 +700,11 @@ def traverse_related_sql(self, model: BaseModel, alias: str, query: Query) -> tu
         assert self.related and not self.store
 
         if self.compute_sudo:
-            model = model.sudo()
-        *path_fnames, last_fname = self.related.split('.')
-        for path_fname in path_fnames:
-            path_field = model._fields[path_fname]
-            if path_field.type != 'many2one':
-                raise ValueError(f'Cannot convert {self} (related={self.related}) to SQL because {path_fname} is not a Many2one')
-            model, alias = path_field.join(model, alias, query)
-
-        return model, model._fields[last_fname], alias
+            table = table._sudo()
+        sql = table
+        for fname in self.related.split('.'):
+            sql = sql[fname]
+        return sql
```

Comment:
Wait, this changes the semantics of the method.  It shouldn't return the result, but a `TableSQL` with the last `field`.